File: src/main/java/com/matamercer/microblog/models/repositories/searches/PostSearch.java
package com.matamercer.microblog.models.repositories.searches;

import com.matamercer.microblog.models.entities.Blog;
import com.matamercer.microblog.models.entities.PostTag;
import com.matamercer.microblog.models.enums.PostCategory;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
public class PostSearch {
    private Blog blog;
    private PostCategory postCategory;
    private Set<PostTag> postTags;
}

File: src/main/java/com/matamercer/microblog/models/repositories/searches/PostSearch.kt
package com.matamercer.microblog.models.repositories.searches

import com.matamercer.microblog.models.entities.Blog
import com.matamercer.microblog.models.enums.PostCategory
import com.matamercer.microblog.models.entities.PostTag

class PostSearch {
    var blogs: Set<Blog>? = null
    var postCategory: PostCategory? = null
    var postTags: Set<PostTag>? = null
}